docs/functionalabstractions/concretetypes/index.md

```diff
@@ -3,4 +3,111 @@ id: index
+
+This is quite an important property for being able to build more complex structures from simpler ones. Otherwise the result of sequentially building more complex structures from simpler ones is not well defined because it depends on the order of operations and there are an exponentially increasing number of ways we could order these operations as we combine more and more values.
+
+There are also an extremely large number of data types that can be combined using associative operations.
+
+For example integers can be combined associatively using addition and multiplication, among other operations. String concatenation is also associative, as is concatenation of lists of any concrete type.
+
+For these reasons the `Associative` abstraction forms the root of the set of abstractions for describing ways to combine concrete types in ZIO Prelude.
+
+### Commutative
+
+The `Commutative` abstraction builds on the `Associative` abstraction by describing a combining operation that is not only associative but also commutative.
+
+```scala
+a <> b === b <> a
+```
+
+This means that not only does the order of operations not matter, but the order in which we combine the values also does not matter.
+
+Commutativity is a powerful property that gives us additional ability to reason about our code. For instance, if the values we are combining are in an arbitrary order, say because they are being produced asynchronously by multiple concurrent processes, then commutativity guarantees that we will get the same result no matter what order we combine them in.
+
+Commutativity is a stronger property than associativity but that also means there are fewer ways of combining values that are commutative than there are ways of combining that are associative. For example, integer addition and multiplication are commutative but string concatenation is not.
+
+### Identity
+
+The `Identity` abstraction builds on the `Associative` abstraction in a different way. It describes a combining operation that is not only associative but also has an identity element.
+
+```scala
+a <> identity === a
+identity <> a === a
+```
+
+This identity element is a "neutral value" that can be combined with any other value an unlimited number of times with the combining function and does not change the other value.
+
+For example `0` is an identity element with respect to addition, `1` is an identity element with respect to multiplication, the empty string is an identity element with respect to string concatenation, and the empty list is an identity element with respect to list concatenation.
+
+An identity element gives us a "starting point" for combining values that we know will never change the value. It also tells us when we can safely eliminate a value from the combining operation because we know the value will never change the result.
+
+### Inverse
+
+The `Inverse` abstraction further builds on the `Identity` abstraction with the concept of an inverse operation.
```

Comment:
I think we need more explanation here along the lines of source code of the same that justifies the view of this being a binary operator rather than unary as the math analog is unary and so a first reading is confusing.